addons/stock/wizard/stock_generate_serial.py

```diff
@@ -8,7 +8,7 @@ class StockGenerateSerial(models.TransientModel):
 
     move_id = fields.Many2one('stock.move')
     next_serial = fields.Char('First SN')
-    next_serial_count = fields.Integer('Number of SN')
+    next_serial_count = fields.Integer('Number of SN', related='move_id.next_serial_count')
```

Comment:
this should be making the field:

* store=False which we don't do in stable, you should add store=True if you change the field in related
*  prevent setting it manually while before it was settable manually (it would be restorable with readonly=False attribute) but functionally maybe we don't want it to be settable